### PR TITLE
Artp 373/reduce theme complexity

### PR DESCRIPTION
**Pull request by keshaldesai**
Updates:
* Reduce theme complexity
* Generalize colors into names (e.g. 'lightBackground')
* Switch to styled-components (we get full benefits of v4 including dynamic global styles, and direct refs)
* Remove emotion
* Remove react-app-rewired
* Remove extra babel packages

TODO:
* Better types
* Simplify theme creation (will require heavy work around style guide)
* Once theme creation is simplified, consolidate colors into a defaultTheme file

### CODE REVIEW

File: src/client/package.json [hunk 1]
```diff
@@ -92,14 +88,12 @@
     "reselect": "3.0.1",
     "ress": "^1.2.2",
     "rxjs": "6.1.0",
+    "styled-components": "^4.1.3",
```
Comment (snony) [inline src/client/package.json]: Are we using this one (styled-components) or just react-emotion?

File: src/client/src/routes/OrderTicketRoute/Notification.tsx [hunk 2]
```diff
@@ -107,7 +108,7 @@ Position.defaultProps = {
   position: 'bottom',
 }
 
-const Transistion: Styled<{ visible?: boolean; position?: 'top' | 'bottom' }> & { [k: string]: any } = styled(Block)`
+const Transistion: any = styled(Block)<{ visible?: boolean; position?: 'top' | 'bottom' }>`
```
Comment (snony) [inline src/client/src/routes/OrderTicketRoute/Notification.tsx]: Based on @bhavesh-desai-scratch :  we should Type every variable and avoid using any type

File: src/client/src/routes/OrderTicketRoute/TextField.tsx [hunk 3]
```diff
@@ -93,10 +93,10 @@ export const Input = styled.input`
   width: 100%;
   max-width: 100%;
 
-  ${curryProps(mapColorProps, { fg: 'secondary.base' })};
+  ${curryProps<any>(mapColorProps, { fg: 'secondary.base' })};
```
Comment (snony) [inline src/client/src/routes/OrderTicketRoute/TextField.tsx]: type this one maybe?

File: src/client/src/routes/StyleguideRoute/styled/Color.tsx [hunk 4]
```diff
@@ -13,8 +13,8 @@ export interface ColorProps {
 export const colorProps: MappedPropMap<ColorProps> = {
   backgroundColor: v => (colorProps.bg as MappedPropFn<ColorProps>)(v),
   textColor: v => (colorProps.fg as MappedPropFn<ColorProps>)(v),
-  bg: ({ theme, backgroundColor, bg = backgroundColor }) => bg && { backgroundColor: getColor(theme, bg) },
-  fg: ({ theme, color, textColor, fg = textColor || color }) => fg && { color: getColor(theme, fg) },
+  //bg: ({ theme, backgroundColor, bg = backgroundColor }) => bg && { backgroundColor: getColor(theme, bg) },
```
Comment (snony) [inline src/client/src/routes/StyleguideRoute/styled/Color.tsx]: Perhaps, remove this if it is no longer needed.

File: src/client/src/routes/OrderTicketRoute/Notification.tsx (hunk 2 above)
Comment (keshaldesai) [reply src/client/src/routes/OrderTicketRoute/Notification.tsx]: Unfortunately typescript has an issue with using the `Component.from` syntax so I had to use any here. I think this component should just be rewritten in the future to use `keyframes`.

**Comment by keshaldesai** [comment]
@danmcdaid You can still have nested/child theming. Example: `/src/client/src/ui/status-bar/StatusBar.tsx`

By just using the `ThemeProvider` from `styled-components`, You can pull off the old theme using `withTheme`, or you can just supply an object.

Unfortunately, you will lose the typing of the selected or new theme. However, we did not have typing using the old method anyways. We would need to make a new `styled` with the type of the nested theme and only use that to create styled components after that point.

**Pull request merged**